src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt

```diff
@@ -96,9 +111,21 @@ class Cargo(
 
     @Throws(ExecutionException::class)
     fun reformatFile(owner: Disposable, file: VirtualFile, listener: ProcessListener? = null): ProcessOutput {
+
+        val arguments = mutableListOf("--all", "--", file.path)
+
+        var (emit, skip_children) = checkSupportForRustfmtFlags()
+        if (!emit) {
+            arguments += "--write-mode=overwrite"
+        }
```

Comment:
I'd prefer to pass `--emit=files` explicitly to commit our intention
It's true that it's current default value but default value can be changed in the future and it will change our behavior too (probably in an unexpected way)